src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -101,18 +100,25 @@ export const Deposit: React.FC<DepositProps> = ({
 
   const getDepositGasEstimate = useCallback(
     async (deposit: DepositValues): Promise<string | undefined> => {
-      if (!(userAddress && assetReference && yearnInvestor && accountId && opportunityData)) return
+      if (!(userAddress && assetReference && accountId && opportunityData)) return
       try {
-        const yearnOpportunity = await yearnInvestor.findByOpportunityId(opportunityData.assetId)
-        if (!yearnOpportunity) throw new Error('No opportunity')
-        const preparedTx = await yearnOpportunity.prepareDeposit({
-          amount: bnOrZero(deposit.cryptoAmount).times(`1e+${asset.precision}`).integerValue(),
-          address: fromAccountId(accountId).account,
-        })
-        // TODO(theobold): Figure out a better way for the safety factor
-        return bnOrZero(preparedTx.gasPrice)
-          .times(preparedTx.estimatedGas)
-          .integerValue()
+        const amountCryptoBaseUnit = bnOrZero(deposit.cryptoAmount).times(
+          bn(10).pow(asset.precision),
+        )
+        const quote = await getThorchainSaversQuote(asset, amountCryptoBaseUnit)
+        const chainAdapters = getChainAdapterManager()
+        const adapter = chainAdapters.get(chainId) as unknown as UtxoBaseAdapter<UtxoChainId>
+        const fee = (
+          await adapter.getFeeData({
+            to: quote.inbound_address,
+            value: amountCryptoBaseUnit.toFixed(0),
+            chainSpecific: { pubkey: userAddress, from: '' },
+            sendMax: false,
+          })
+        ).fast.txFee
+        // We might need a dust reconciliation Tx for UTXOs, so we assume gas * 2
+        return bnOrZero(fee)
+          .times(isUtxoChainId(chainId) ? 2 : 1)
```

Comment:
We won't need the dust-like flow as we're going broadcast both Txs within one block (see https://github.com/shapeshift/web/pull/3640), but that's still two Txs, so this double gas is still correct